### PR TITLE
Test enhancements

### PR DESCRIPTION
**Pull request by dentarg**
I was going to do #11, but...

### CODE REVIEW

**Comment by dentarg** [comment]
With [rocha](https://github.com/bahmutov/rocha) we can run the tests in random order:

    rocha test/test_*.js

(`npm install -g rocha` if you want to try it)

but then we loose the ability to pass on arguments to `npm test`, e.g.

    npm test test/test_query

**Comment by walro** [comment]
Such DRY, such contexts, such describes. :star2:

File: test/test_client.js [hunk 1]
```diff
@@ -1,66 +1,91 @@
...
         });
     });
 
-    it('endpoint url', function(){
+    describe('#endpoint_url', function(){
         var c = new Client();
-        expect(c.endpointUrl()).to.be.equal(c.BASE_URL + c.SEARCH_PATH)
+        var expected = c.BASE_URL + c.SEARCH_PATH;
+
+        it('should eq '+expected, function() {
```
Comment (walro) [inline test/test_client.js]: Intenetionally not written as ` + expected`?

Comment (walro) [inline test/test_client.js]: Hehe cant edit on phone, intentionally*

Comment (dentarg) [inline test/test_client.js]: Nope, just didn't think about it

we also mix `function() {` and `function(){`, I think we should address all these style issues separately

**Comment by dentarg** [comment]
> Unfortunately only randomizes per file (in test/test_*.js).

That could be solved with a wrapper, I guess, or perhaps https://github.com/mochajs/mocha/wiki/Using-mocha-programmatically.

I think it's good enough for now.

**Comment by walro** [comment]
I browsed through the last three commits and checked the output on Travis, it's definitely an improvement and nice addition of UTC tests. Ship it!

**Comment by dentarg** [comment]
thx 💕

**Pull request merged**